services/web-app/utils/use-query-state.js

```diff
@@ -75,19 +66,21 @@ export const useQueryState = (
 
       // Don't rely on router.query here because that would cause unnecessary renders when other
       // query parameters change
-      const query = new URLSearchParams(window.location.search)
+      const query = queryString.parseUrl(window.location.search, {
+        ...queryStringConfig,
+        parseNumbers,
+        parseBooleans
+      }).query
 
-      if (newValue === null || newValue === undefined) {
-        query.delete(key)
-      } else {
-        query.set(key, serialize(newValue))
-      }
+      query[key] = newValue
 
-      router.replace(`?${query.toString()}`, undefined, { shallow: true, scroll: false })
+      // Change query state without rerendering page
+      history.replaceState(null, null, `?${queryString.stringify(query, queryStringConfig)}`)
+
+      setValue(getValue())
     },
-    // The Next.js router updates `router.replace`, which should not trigger re-renders
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [getValue, key, serialize]
+    [getValue, key]
```

Comment:
same question here about which deps are missing

Reply:
`parseNumbers`, `parseBooleans`. Should I add those to the deps array or continue to ignore them? I get a little confused about adding deps I don't really "care" about to the deps array

Reply:
Yeah, definitely add those. In general, it's best to be exhaustive with those deps since it helps react make better determinations about when to re-execute hooks. Effect hooks aren't so much about defining the deps you want to cause rerenders; they're instead about tracking dependencies and allowing react to take over from there. In the docs I think they say someplace that the effect hook deps list will potentially at some point go away and be generated programmatically, so it's best for us to be in alignment with their eslint rules in all possible places.

Reply:
Done!